[jodielzy] iP

jodielzy opened the pull request:
# 🚀 Lucy Task Manager

> “Your mind is for having ideas, not holding them.” – *David Allen* ([source](https://gettingthingsdone.com/))

Lucy is a simple, text-based task manager that helps you **organize** and **track** your tasks efficiently.

## Features 📝
- ✅ Manage tasks, deadlines, and events
- 🔍 **Find tasks** using keywords
- 💾 Persistent storage for tasks
- 🎯 User-friendly **command-line interface**

## How to Use 🛠️
1. **Download** the latest release from [GitHub Releases](https://github.com/your-repo/releases).
2. **Run** the application:
    ```sh
    java -jar lucy.jar
    ```
3. **Add tasks** with the following commands:
    - `todo read book`
    - `deadline return book /by 2024-06-06`
    - `event attend meeting /from 10AM /to 12PM`
4. **Find tasks** using:
    ```sh
    find book
    ```
5. **Mark tasks as done:**
    ```sh
    mark 1
    ```
6. **Exit the application**:  
   ```sh
   bye

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

jodielzy committed:
Added Lucy.java

jodielzy committed:
Modified Lucy.java

jodielzy committed:
Modified Lucy.java

jodielzy committed:
Modified Lucy.java, Added class Task

jodielzy committed:
Modified Lucy.java

jodielzy committed:
Modified Lucy.java, Added Task.java, Deadline.java, Todo.java, Event.java, Updated input.txt and EXPECTED.txt

jodielzy committed:
Added LucyException.java

jodielzy committed:
Modified Lucy.java

jodielzy committed:
Modified Lucy.java

jodielzy committed:
Add TaskManager.java

jodielzy committed:
Modify Deadline.java

jodielzy committed:
Modify Event.java

jodielzy committed:
Modify Lucy.java

jodielzy committed:
Modify Task.java

jodielzy committed:
Modify TaskManager.java

jodielzy committed:
Modify Todo.java

jodielzy committed:
Merge branch 'branch-Level-7'

jodielzy committed:
Modify Deadline.java

jodielzy committed:
Modify Lucy.java

jodielzy committed:
Modify TaskManager.java

jodielzy committed:
Merge branch 'branch-Level-8'

jodielzy committed:
Add Parser.java

jodielzy committed:
Modify Lucy.java

jodielzy committed:
Modify Parser.java

jodielzy committed:
Delete TaskManager.java

jodielzy committed:
Add Storage.java

jodielzy committed:
Add TaskList.java

jodielzy committed:
Add Ui.java

jodielzy committed:
Merge branch 'branch-A-MoreOOP'

jodielzy committed:
Refractor all classes into package lucy

jodielzy committed:
Merge branch 'add-gradle-support'

jodielzy committed:
Add ParserTest.java

jodielzy committed:
Modify ParserTest.java

jodielzy committed:
Add TaskListTest.java

jodielzy committed:
Modify TaskList.java

jodielzy committed:
Set up Gradle to run test

jodielzy committed:
Change main class under applications

jodielzy committed:
Add header comments

jodielzy committed:
Modify TaskListTest.java

jodielzy committed:
Modify Lucy.java

jodielzy committed:
Modify TaskList.java

jodielzy committed:
Merge branch 'branch-A-JavaDoc'

jodielzy committed:
Merge branch 'branch-A-CodingStandard'

jodielzy committed:
Merge branch 'branch-Level-9'

e1093051 commented on src/main/java/lucy/Ui.java:
```diff
@@ -0,0 +1,58 @@
+    }
+
+    /**
+     * Start of an error message.
```
Would it be better to change it to something like "Displays..."

e1093051 commented on src/main/java/lucy/Lucy.java:
```diff
@@ -0,0 +1,112 @@
+ */
+public class Lucy {
+    private Storage storage;
+    private TaskList tasks;
```
I love your naming for variables -- they explain themselves

e1093051 commented on src/main/java/lucy/TaskList.java:
```diff
@@ -0,0 +1,104 @@
+     * @param storage The storage instance to save tasks.
+     * @throws LucyException If the index is out of range.
+     */
+    public void markTask(int index, boolean done, Storage storage) throws LucyException {
```
Maybe can change done to isDone.

samuelkwik replied:
Agreed with other comment, done could perhaps be isDone?

e1093051 commented on src/main/java/lucy/Ui.java:
```diff
@@ -0,0 +1,58 @@
+
+    /**
+     * Start of an error message.
+     * @param message Displays all error messages starting with "OOPS!!!"
```
Maybe can change the explanation a bit to make it more about explaining the param message(?)

leewaikin19 commented on src/main/java/lucy/TaskList.java:
```diff
@@ -0,0 +1,104 @@
+        }
+        return tasks.get(index);
+    }
+    public void findTasks(String keyword) {
```
Maybe a better name would be `printMatchingTasks` instead? `findTasks` suggests that it would return a list of tasks

leewaikin19 commented on src/main/java/lucy/Lucy.java:
```diff
@@ -0,0 +1,112 @@
+                        }
+                        break;
+                    case "event":
+                        String[] eventParts = parsedCommand[1].split(" /from | /to ");
```
Instead of `eventParts`, maybe a better name would be `eventArguments` or `eventArgs` or even `eventDetails`

leewaikin19 commented on src/main/java/lucy/Lucy.java:
```diff
@@ -0,0 +1,112 @@
+                        tasks.listTasks();
+                        break;
+                    case "mark":
+                        int markIndex = Integer.parseInt(parsedCommand[1]) - 1;
```
Perhaps a better name would be `taskIndexToMark`

leewaikin19 commented on src/main/java/lucy/Lucy.java:
```diff
@@ -0,0 +1,112 @@
+                        ui.showMessage("Nice! I've marked this task as done.");
+                        break;
+                    case "unmark":
+                        int unmarkIndex = Integer.parseInt(parsedCommand[1]) - 1;
```
Same comment here, maybe `taskIndexToUnmark` would be better

leewaikin19 reviewed (commented):
Great overall naming schemes!

samuelkwik commented on src/main/java/lucy/TaskList.java:
```diff
@@ -0,0 +1,104 @@
+     */
+    public void addTask(Task task, Storage storage) {
+        tasks.add(task);
+        storage.saveTasks(tasks);
```
Just nitpicking: what happens if the `storage.saveTasks(tasks);` throws an `IOException`, will the task still be added to the `TaskList`?

samuelkwik commented on src/main/java/lucy/Ui.java:
```diff
@@ -0,0 +1,58 @@
+     * Displays the welcome message.
+     */
+    public void showWelcome() {
+        System.out.println("_________________________________");
```
Maybe you could use `showLine()` instead?

samuelkwik commented on src/main/java/lucy/Storage.java:
```diff
@@ -0,0 +1,96 @@
+                String[] parts = fileScanner.nextLine().split("\\s*\\|\\s*");
+                Task task;
+                switch (parts[0]) {
+                    case "T":
```
I believe the coding standard states that the case keyword should be at same indentation as switch?

samuelkwik commented on src/main/java/lucy/TaskList.java:
```diff
@@ -0,0 +1,104 @@
+            throw new LucyException("lucy.Task index out of range.");
+        }
+        if (done) {
+            tasks.get(index).markAsDone();
```
Just a suggestion: Why not use your `this.getTasks()` here? This way, you can rely on getTasks() to catch the index out of range error without repetition.

samuelkwik commented on src/main/java/lucy/TaskList.java:
```diff
@@ -0,0 +1,104 @@
+    }
+
+    /**
+     * Retrives a task from the list based on its index.
```
I think there's a typo? Retrieves instead of Retrives.

samuelkwik commented on src/main/java/lucy/Deadline.java:
```diff
@@ -0,0 +1,39 @@
+     * @return The formatted string to be saved in a file.
+     */
+    @Override
+    public String toFileFormat() {
```
I like your method naming! Concise but still clear. 😄

samuelkwik reviewed (commented):
Other than some nits, code is well-structured and clear. LGTM 👍.

jodielzy committed:
Add DialogBox.fxml, MainWindow.fxml

jodielzy committed:
Add Profile Images for Chatbox

jodielzy committed:
Add DialogBox.java, Launcher.java, MainWindow.java

jodielzy committed:
Add Main.java

jodielzy committed:
Modify Lucy.java, TaskList.java

jodielzy committed:
Modify build.gradle

jodielzy committed:
Merge branch 'branch-Level-10'

jodielzy committed:
Delete Ui.java

jodielzy committed:
Update main class

jodielzy committed:
Modify build.gradle to enable assertion

jodielzy committed:
Modify Lucy.java

jodielzy committed:
Modify Parser.java

jodielzy committed:
Modify Storage.java

jodielzy committed:
Modify TaskList.java

jodielzy committed:
Modify TaskList.java

Removed findTasks method as it is not used. There was a duplicated
method findTasksString.

Removed listTasks() method as it is not used. There was a duplicated
method listTasksString.

jodielzy committed:
Merge pull request #2 from jodielzy/branch-A-Assertions

branch-A-Assertions

jodielzy committed:
Merge branch 'master' into branch-A-CodeQuality

jodielzy committed:
Merge pull request #3 from jodielzy/branch-A-CodeQuality

branch-A-CodeQuality

jodielzy committed:
Modify Lucy.java

Method getResponse() is too long. Extracted each possible case into
seperated methods for better readability.

jodielzy committed:
Modify Storage.java

Method loadTasks() is too long. Extracted some code blocks into separate
methods for better readability.

jodielzy committed:
Modify Deadline.java

jodielzy committed:
Modify Event.java

jodielzy committed:
Modify Lucy.java

jodielzy committed:
Modify Main.java

jodielzy committed:
Modify Task.java

jodielzy committed:
Modify TaskList.java

jodielzy committed:
Modify Todo.java

jodielzy committed:
Merge branch 'branch-C-Undo'

jodielzy committed:
Add header comments to all methods and classes

jodielzy committed:
Add dialog-box.css and main.css

jodielzy committed:
Modify DialogBox.java

jodielzy committed:
Modify Lucy.java

jodielzy committed:
Modify Main.java

jodielzy committed:
Modify dialog-box.css

jodielzy committed:
Modify main.css

jodielzy committed:
Modify DialogBox.fxml

jodielzy committed:
Modify MainWindow.fxml

jodielzy committed:
Merge branch 'branch-A-BetterGui'

jodielzy committed:
Modify Lucy.java

jodielzy committed:
Modify TaskList.java

jodielzy committed:
Modify README.md

jodielzy committed:
Modify Lucy.java

jodielzy committed:
Add Ui.png

jodielzy committed:
Modify README.md

jodielzy committed:
Modify Deadline.java, Event.java, Lucy.java, Task.java, TaskList.java
